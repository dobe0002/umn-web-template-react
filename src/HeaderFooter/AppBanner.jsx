/* eslint-disable react/prop-types */
/* eslint-disable no-dupe-keys */
import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

const bannerCSS = {
  paddingLeft: '10px',
  paddingTop: '8px',
  minHeight: '52px',
  borderBottom: ' #f93 2px solid',
  fontFamily: '"Trebuchet MS", Arial, Helvetica, sans-serif',
  fontSize: '35px',
  fontWeight: 'bold',
  color: '#5b0013',
  marginTop: 0,
  marginBottom: 0,
  background: '#f0e9d1' /* Old browsers */,
  background:
    '-moz-linear-gradient( top, #f0e9d1 0%, #ffde7c 100%)' /* FF3.6-15 */,
  background:
    '-webkit-linear-gradient( top, #f0e9d1 0%, #ffde7c 100%)' /* Chrome10-25,Safari5.1-6 */,
  background:
    'linear-gradient( to bottom, #f0e9d1 0%,#ffde7c 100%)' /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */,
};
const AppBanner = (props) => {
  const { appTitle } = props;
  return (
    <div role='contentinfo' style={bannerCSS}>
      {appTitle}
    </div>
  );
};
// AppBanner.propTypes = {
//   appTitle: PropTypes.string.isRequired,
// };

export default AppBanner;
