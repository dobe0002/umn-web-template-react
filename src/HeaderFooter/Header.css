.umnHeaderWrapperDiv {
  position: relative;
  min-height: 68px;
  background-color: #7a0019;
  padding: 0 54px;
  padding-top: 15px;
  color: #fff;
  font-size: 14px;
  font-family: Helvetica, 'Helvetica Neue', Futura, Verdana, Geneva, sans-serif;
  box-sizing: border-box;
}

@media (max-width: 609px) {
  .umnHeaderWrapperDiv {
    min-height: 100px;
  }
  .umnHeaderSearchForm {
    display: block;
    right: auto;
    top: 64px;
  }
}
@media (max-width: 775px) {
  .umnHeaderSearchForm {
    right: 17px;
  }
}

.umnHeaderSearchForm {
  display: block;
  position: absolute;
  right: 54px;
  bottom: 8px;
}

.umnHeaderSearchInput {
  font-size: 14px;
  border: none;
  padding: 4px;
  width: 175px;
}
.umnHeaderSearchButton {
  background-color: #7a0019;
  border: none;
}
