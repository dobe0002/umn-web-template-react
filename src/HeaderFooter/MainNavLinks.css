.umnHeaderMainNavLinksUl,
.umnHeaderMainNavLinksUl-header {
  margin: 0;
  list-style-type: none;
  display: none;
}

.umnHeaderMainNavLinksUl-header {
  position: absolute;
  right: 54px;
  top: 8px;
}

@media (max-width: 775px) {
  .umnHeaderMainNavLinksUl-header {
    display: none;
  }
  .umnHeaderMainNavLinksUl {
    display: block;
  }
}
@media (min-width: 776px) {
  .umnHeaderMainNavLinksUl-header {
    display: block;
  }
  .umnHeaderMainNavLinksUl {
    display: none;
  }
}

.umnHeaderMainNavLinksLi {
  margin: 0;
  display: inline-block;
  margin-left: 1.25rem;
}

.umnHeaderMainNavLinksA,
.umnHeaderMainNavLinksA-header {
  text-decoration: none;
  color: #404d5b;
}
.umnHeaderMainNavLinksA-header {
  color: #fff;
}
.umnHeaderMainNavLinksA:hover,
.umnHeaderMainNavLinksA:focus {
  text-decoration: underline;
  color: #404d5b;
}
.umnHeaderMainNavLinksA-header:hover,
.umnHeaderMainNavLinksA-header:focus {
  text-decoration: underline;
  color: #fff;
}
