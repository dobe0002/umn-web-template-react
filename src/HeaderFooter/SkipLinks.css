.umnHeaderSkipLinksWrapperDiv {
  position: absolute;
  top: 0;
  left: -10000px;
  width: 1px;
  height: 1px;
  overflow: hidden;
  padding: 15px;
  background: #fff;
  border-radius: 0 0 5px 5px;
  box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.5);
}

.umnHeaderSkipLinksWrapperDiv:focus-within {
  left: 0;
  width: auto;
  height: auto;
  overflow: auto;
}

.umnHeaderSkipLinksUl {
  margin: 0;
  padding: 0;
}

.umnHeaderSkipLinksLi {
  display: inline-block;
  margin-right: 15px;
}
