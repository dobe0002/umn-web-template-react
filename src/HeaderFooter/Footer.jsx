/* eslint-disable import/extensions */
/* eslint-disable react/prop-types */
import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */
import MainNavLinks from './MainNavLinks';

const defaultYear = '2020';
const footerCSS = {
  fontFamily:
    'Helvetica, "Helvetica Neue", Futura, Verdana, Geneva, sans-serif',
  padding: '20px 54px',
  textAlign: 'center',
  fontSize: '0.8rem',
  color: '#000',
  borderTop: '1px solid #f0f0f0',
};
const linkCSS = {
  color: '#404d5b',
  textDecoration: 'underline',
};

const Footer = (props) => {
  const { year, label } = props;
  const displayYear = year === '' || year === undefined ? defaultYear : year;
  const displayLabel =
    label === '' || label === undefined
      ? 'University copyright information'
      : label;
  return (
    <footer role='contentinfo' style={footerCSS} aria-label={displayLabel}>
      <MainNavLinks mode='footer' />
      <p>
        &copy;
        <span className='year'>{displayYear}</span>
        <span> </span>
        Regents of the University of Minnesota. All rights reserved. The
        University of Minnesota is an equal opportunity educator and employer.
        <span> </span>
        <a href='http://privacy.umn.edu' style={linkCSS}>
          Privacy Statement
        </a>
      </p>
    </footer>
  );
};
// Footer.propTypes = {
//   year: PropTypes.string,
//   label: PropTypes.string
// };
// Footer.defaultProps = {
//   year: '2020',
//   label: 'University copyright information'
// };

export default Footer;
