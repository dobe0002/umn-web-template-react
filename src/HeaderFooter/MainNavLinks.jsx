/* eslint-disable react/prop-types */
/* eslint-disable import/extensions */
import React from 'react';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

import LockIcon from './icons/LockIcon.jsx';

import './MainNavLinks.css'; /* Need to make a separate CSS  because it uses media queries */

const MainNavLinks = (props) => {
  const { mode } = props;

  const displayMode = mode === undefined ? 'header' : mode;
  const ulClassName =
    displayMode === 'header'
      ? 'umnHeaderMainNavLinksUl-header'
      : 'umnHeaderMainNavLinksUl';

  const aClassName =
    displayMode === 'header'
      ? 'umnHeaderMainNavLinksA-header'
      : 'umnHeaderMainNavLinksA';
  return (
    <ul className={ulClassName}>
      <li className='umnHeaderMainNavLinksLi'>
        <a href='http://onestop.umn.edu/' className={aClassName}>
          One Stop
        </a>
      </li>
      <li className='umnHeaderMainNavLinksLi'>
        <a href='https://www.myu.umn.edu/' className={aClassName}>
          <span>MyU </span>
          <LockIcon color={displayMode === 'header' ? '#fff' : '#404d5b'} />
        </a>
      </li>
    </ul>
  );
};

export default MainNavLinks;

// MainNavLinks.propTypes = {
//   mode: PropTypes.string,
// };
// MainNavLinks.defaultProps = {
//   mode: 'header',
// };
