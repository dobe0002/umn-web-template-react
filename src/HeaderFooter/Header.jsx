/* eslint-disable react/prop-types */
/* eslint-disable import/extensions */
import React from 'react';
// import PropTypes from 'prop-types';
import WordMark from './icons/WordMark';
import MainNavLinks from './MainNavLinks';
import SearchIcon from './icons/SearchIcon';
import SkipLinks from './SkipLinks';
import AppBanner from './AppBanner';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

import './Header.css'; /* Need to make a separate CSS  because it uses media queries */

const Header = (props) => {
  const { mainContentId, mainNavId, appTitle } = props;
  return (
    <header>
      <div className='umnHeaderWrapperDiv'>
        <SkipLinks mainContentId={mainContentId} mainNavId={mainNavId} />
        <div>
          <a href='http://umn.edu/'>
            <WordMark />
          </a>
          <MainNavLinks color='#fff' />
        </div>
        <form
          className='umnHeaderSearchForm'
          id='search-form'
          name='gs'
          action='https://usearch.umn.edu/'
        >
          <input
            className='umnHeaderSearchInput'
            type='text'
            name='query'
            maxLength='256'
            placeholder='Search'
            aria-label='Search for'
          />

          <button
            className='umnHeaderSearchButton'
            type='submit'
            name='submit'
            value='search'
            aria-label='Submit Search'
          >
            <SearchIcon />
          </button>
        </form>
      </div>
      {appTitle !== undefined && appTitle !== '' && (
        <AppBanner appTitle={appTitle} />
      )}
    </header>
  );
};

// Header.propTypes = {
//   mainNavId: PropTypes.string,
//   mainContentId: PropTypes.string,
//   appTitle: PropTypes.string,
// };
// Header.defaultProps = {
//   mainNavId: undefined,
//   mainContentId: undefined,
//   appTitle: undefined,
// };

export default Header;
