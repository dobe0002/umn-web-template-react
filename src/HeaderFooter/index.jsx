/* eslint-disable import/extensions */
import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header.jsx';
import Footer from './Footer.jsx';

const UmnHeader = (props) => {
  const { mainNavId, mainContentId, appTitle } = props;
  return (
    <Header
      mainNavId={mainNavId}
      mainContentId={mainContentId}
      appTitle={appTitle}
    />
  );
};
UmnHeader.propTypes = {
  mainNavId: PropTypes.string,
  mainContentId: PropTypes.string,
  appTitle: PropTypes.string,
};
UmnHeader.defaultProps = {
  mainNavId: undefined,
  mainContentId: undefined,
  appTitle: undefined,
};

/* *********************************************************************************************************** */
const UmnFooter = (props) => {
  const { year } = props;
  return <Footer year={year} />;
};

UmnFooter.propTypes = {
  year: PropTypes.string,
};
UmnFooter.defaultProps = {
  year: undefined,
};

export { UmnHeader, UmnFooter };
