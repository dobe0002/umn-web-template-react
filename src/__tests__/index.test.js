/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { UmnHeader, UmnFooter } from '../HeaderFooter/index';

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

configure({ adapter: new Adapter() });

describe('Index - entry point', () => {
  it('Accessibility check', async () => {
    const umnTemplate = mount(
      <div>
        <UmnHeader />
        <UmnFooter />
      </div>
    );
    const results = await axe(umnTemplate.getDOMNode());

    // console.log('Axe violations', results.violations);
    expect(results).toHaveNoViolations();
  });
  it('All items renders correctly', () => {
    const umnTemplate = mount(
      <div>
        <UmnHeader />
        <UmnFooter />
      </div>
    );
    // console.log(header.html());
    expect(umnTemplate).toMatchSnapshot();
  });
});
