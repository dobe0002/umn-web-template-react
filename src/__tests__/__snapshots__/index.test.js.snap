// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index - entry point All items renders correctly 1`] = `ReactWrapper {}`;
